spring-integration-ip/src/test/java/org/springframework/integration/ip/dsl/IpIntegrationTests.java

```diff
@@ -128,6 +128,13 @@ public void testUdp() {
 		assertEquals("foo", Transformers.objectToString().transform(received).getPayload());
 	}
 
+	@Test
+	public void testUdpInheritance() {
+		Udp.outboundMulticastAdapter("headers['udp_dest']")
+			.lengthCheck(true)
+			.timeToLive(10);
+	}
```

Comment:
OK. I'll just add in the end an assertion  that we have proper `instanceOf` after `Spec` evaluation.